packages/@aws-cdk/aws-applicationautoscaling/lib/target-tracking-scaling-policy.ts

```diff
@@ -0,0 +1,175 @@
+import cloudwatch = require('@aws-cdk/aws-cloudwatch');
+import cdk = require('@aws-cdk/cdk');
+import { cloudformation } from './applicationautoscaling.generated';
+import { ScalableTarget } from './scalable-target';
+
+/**
+ * Base interface for target tracking props
+ *
+ * Contains the attributes that are common to target tracking policies,
+ * except the ones relating to the metric and to the scalable target.
+ *
+ * This interface is reused by more specific target tracking props objects
+ * in other services.
+ */
+export interface BaseTargetTrackingProps {
+  /**
+   * A name for the scaling policy
+   *
+   * @default Automatically generated name
+   */
+  policyName?: string;
+
+  /**
+   * Indicates whether scale in by the target tracking policy is disabled.
+   *
+   * If the value is true, scale in is disabled and the target tracking policy
+   * won't remove capacity from the scalable resource. Otherwise, scale in is
+   * enabled and the target tracking policy can remove capacity from the
+   * scalable resource.
+   *
+   * @default false
+   */
+  disableScaleIn?: boolean;
+
+  /**
+   * Period after a scale in activity completes before another scale in activity can start.
+   *
+   * @default No scale in cooldown
+   */
+  scaleInCooldownSec?: number;
+
+  /**
+   * Period after a scale out activity completes before another scale out activity can start.
+   *
+   * @default No scale out cooldown
+   */
+  scaleOutCooldownSec?: number;
```

Comment:
I totally agree with making `scaleInCooldownSec` and `scaleOutCooldownSec` optional.